felix/iptables/table.go

```diff
@@ -869,6 +869,7 @@ func (t *Table) attemptToGetHashesAndRulesFromDataplane() (hashes map[string][]s
 	waitErr := cmd.Wait()
 	if waitErr != nil {
 		log.WithError(waitErr).Warn("iptables save failed")
+		log.Debugf("failing iptables save command is '%s", cmd.String())
```

Comment:
I think for optimization purposes, we should structure this like so:

```go
if log.IsLevelEnabled(log.DebugLevel) {
  log.Debugf(...)
}
```

This ensures we don't do any extra work (e.g., rendering the String() function) when not actually in debug mode.

Reply:
Looking at this a little bit closer, I am not sure there's a whole lot of value to be gained from this log.

Namely, because we always run the following:

```
cmd := t.newCmd(t.iptablesSaveCmd, "-t", t.Name)
```

which turns into

```
iptables-save -t TABLE
```

The only thing that really varies is the table name. I think it would be better to modify the existing warning log to look like this:

```go
log.WithFields(log.Fields{"table": t.Name, "cmd": t.iptablesSaveCmd}).Warn("iptables save failed")